lib/apiservers/engine/backends/portmap/portmap.go

```diff
@@ -108,6 +108,32 @@ func (p *portMapper) MapPort(op Operation, ip net.IP, port int, proto string, de
 	return p.forward(action, ip, port, proto, destIP, destPort, srcIface, destIface)
 }
 
+// iptablesRunAndCheck runs an iptables command with the provided args
+func iptablesRunAndCheck(args []string) error {
+	if output, err := iptables.Raw(args...); err != nil {
+		return err
+	} else if len(output) != 0 {
+		return iptables.ChainError{Chain: "FORWARD", Output: output}
+	}
+	return nil
+}
+
+// iptablesDelete takes the saved args from the Append operation
+// and uses them to delete the previously added rules
+func iptablesDelete(args [][]string) error {
+	var errs []error
+	for _, cmd := range args {
+		cmd[2] = string(iptables.Delete) // change saved action from Append to Delete
```

Comment:
How confident are you there are at least three arguments in cmd? It seems very fragile.